referenceframe/frame.go

```diff
@@ -445,3 +445,96 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(temp)
 }
+
+type sevenDFrame struct {
+	*baseFrame
+	collisionGeometry spatial.Geometry
+}
+
+// New7DFrame created a frame with 7 degrees of freedom.
+func New7DFrame(name string, limits []Limit, collisionGeometry spatial.Geometry) (Frame, error) {
+	if len(limits) != 7 {
+		return nil,
+			errors.Errorf("Must have 7DOF state (x, y, z, OX, OY, OZ, theta) to create 6DFrame, have %d dof", len(limits))
+	}
+
+	return &sevenDFrame{
+		&baseFrame{name, limits},
+		collisionGeometry,
```

Comment:
lets remove the notion of collision here and call this just geometry to conform to what this package cares about